integrations/elasticsearch/tests/test_document_store.py

```diff
@@ -760,3 +810,59 @@ async def test_delete_all_documents_async_no_index_recreation(self, document_sto
         new_doc = Document(id="3", content="New document after delete all")
         await document_store.write_documents_async([new_doc])
         assert await document_store.count_documents_async() == 1
+
+    @pytest.mark.asyncio
+    async def test_delete_by_filter_async(self, document_store):
+        docs = [
+            Document(content="Doc 1", meta={"category": "A"}),
+            Document(content="Doc 2", meta={"category": "B"}),
+            Document(content="Doc 3", meta={"category": "A"}),
+        ]
+        await document_store.write_documents_async(docs)
+        assert await document_store.count_documents_async() == 3
+
+        # Delete documents with category="A"
+        deleted_count = await document_store.delete_by_filter_async(
+            filters={"field": "category", "operator": "==", "value": "A"}
+        )
+        time.sleep(2)  # wait for deletion to be reflected
```

Comment:
Not really as issue, but I would use async sleep here: `await asyncio.sleep(2)`.

Maybe we can add `refresh` support to avoid waiting? This way we could have immediate consistency.